ios/extensions/UIWindow.swift

```diff
@@ -38,7 +38,22 @@ public extension UIWindow {
 
     func getTopWindow() -> UIWindow? {
       // Return the keyboard window if it's available, otherwise return the last window
-      return keyboardWindow ?? UIApplication.shared.windows.last
+      if let keyboardWindow = keyboardWindow {
+        return keyboardWindow
+      }
+      
+      if #available(iOS 13.0, *) {
+        for scene in UIApplication.shared.connectedScenes {
+          if scene.activationState == .foregroundActive,
+            let windowScene = scene as? UIWindowScene,
+            let keyWindow = windowScene.windows.first(where: { $0.isKeyWindow }) {
+            return keyWindow
+          }
+        }
+        return nil
+      } else {
+        return UIApplication.shared.windows.last { $0.isKeyWindow }
+      }
```

Comment:
The changes looks good to me, but may I ask you to slightly re-work this part?

I would like to have this code somewhere isolated, because potentially we may want to get an access to main window in future, so I would move this code to an extension:

```swift
extension UIApplication {
    var activeWindow: UIWindow? {
        if #available(iOS 13.0, *) {
            for scene in connectedScenes {
                if scene.activationState == .foregroundActive,
                   let windowScene = scene as? UIWindowScene,
                   let keyWindow = windowScene.windows.first(where: { $0.isKeyWindow }) {
                    return keyWindow
                }
            }
            return nil
        } else {
            return windows.last { $0.isKeyWindow }
        }
    }
}
```

And then use it like:

```swift
return keyboardWindow ?? UIApplication.shared.activeWindow
```

Not sure if this code works - just wanted to show the idea. Hope you can test these modifications in your project 👀

Reply:
Will try and let you know!

Reply:
@kirillzyusko Looks like it functionally works the same, but I found a new issue that I'm not sure if is the result of this change, or also present in the existing code.

In my app, I have routes guarded behind an auth check, and the `Screens` rendered in the `Navigator` change based on the auth state. When I open the `OverKeyboardView` for the first time, it works, and navigation events such as `push` can be done and `OverKeyboardView` still works. But if the user logs out, the `Screens` are remounted and the next time I try to open this `OverKeyboardView`, the children become invisible again.

Do you think this PR should be put on hold until I figure out the fix for that as well?

Reply:
Ah, sad to hear that it still has issues 😔 

Would you mind to submit a new issues and describe hte problem? Just want to track it somewhere so that I know that I'll not forget about the problem and will return to it :)

Reply:
@kirillzyusko let me create an issue when I get some time this week! Thanks man